### PR TITLE
Pull shared objects out of Hyperdebug Transport Inner member

### PR DESCRIPTION
**Pull request by zhangarino**
This is to remove the cyclic referenced between Hyperdebug Transport Targets and the Inner struct, allowing for Targets to be dropped if they and the Transport object both go out of scope.

This is useful for creating another Transport object after the previous one has been dropped. Otherwise the target objects stay alive forever preventing equivalents from being created the next time a Transport object is created.

### CODE REVIEW

**Review by jesultra — CHANGES_REQUESTED** [review]
Please see my comments on the other review:
https://github.com/lowRISC/opentitan/pull/24118

**Comment by zhangarino** [comment]
Updated with the suggestions from #24118 by just moving the cached references into a new struct that `HyperdebugTransport` directly owns - agreed that the change is much cleaner this way, thanks.

If possible it would be nice to have the change cherry-picked back to earlgrey_es_sival, just because we're currently doing development using that branch. Although the patch is relatively small so it could just be something we maintain on our mirror on top of the upstream.

**Review by jesultra — APPROVED** [review]
Nice, one file changed instead of eight. Now it is straightforward to see everything that this CL does.

I have applied a tag to request automatic cherry-picking to earlgrey_es_sival as soon as this one is merged to master.

I will be on vacation Monday-Wednesday next week, but I want to give the other reviewers a chance to post comments.  I think I will merge this in the evening CET, unless I hear objections.

**Pull request merged**
